.github/workflows/build-python-pdoc.yml

```diff
@@ -0,0 +1,51 @@
+name: DiskANN Build PDoc Documentation
+on: [workflow_call]
+jobs:
+  build-reference-documentation:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python 3.9
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.9
+      - name: "Run Reference Documentation Generation"
+        run: |
+          pip install diskannpy pdoc pipdeptree
+          echo "documentation" > dependencies_documentation.txt
+          pipdeptree >> dependencies_documentation.txt
```

Comment:
Write out the dependencies (including the deps of pdoc) so we can see what deps were used in the build.

Reply:
my concern about `pip install diskannpy` is that it might not actually be showing on pypi.org yet (I don't know for sure, at least).  might be better to download the artifacts built in https://github.com/microsoft/DiskANN/blob/main/.github/workflows/python-release.yml, install the one meant for the python version you specified in the setup-python action, then get it from there - then we won't have to worry about publishing 0.6.0 docs for a 0.6.1 release

Reply:
Added code to build a wheel specifically for the documentation step.